java-frontend/src/main/java/org/sonar/java/model/JavaVersionImpl.java

```diff
@@ -42,19 +44,26 @@ public class JavaVersionImpl implements JavaVersion {
   public static final int MAX_SUPPORTED = JAVA_19;
 
   private final int javaVersion;
+  private final boolean previewFeaturesEnabled;
 
   public JavaVersionImpl() {
     this.javaVersion = -1;
+    this.previewFeaturesEnabled = false;
   }
 
   public JavaVersionImpl(int javaVersion) {
     this.javaVersion = javaVersion;
+    this.previewFeaturesEnabled = false;
+  }
+
+  public JavaVersionImpl(int javaVersion, boolean previewFeaturesEnabled) {
+    this.javaVersion = javaVersion;
+    this.previewFeaturesEnabled = previewFeaturesEnabled;
```

Comment:
This should only be possible on the latest supported version. In our case, if javaVersion >= 19.